src/nix/profile.cc

```diff
@@ -329,7 +329,63 @@ struct CmdProfileInstall : InstallablesCommand, MixDefaultProfile
             manifest.elements.push_back(std::move(element));
         }
 
-        updateProfile(manifest.build(store));
+        try {
+            updateProfile(manifest.build(store));
+        } catch (BuildProfileConflictError & conflictError) {
+            std::string fallbackInstallable = "INSTALLABLE";
+            std::string conflictingInstallable = [&]() {
+                auto maybeConflictingStorePath = store->maybeParseStorePath(conflictError.conflictingPkgPath);
+                if (!maybeConflictingStorePath) {
+                    printError("warning: unable to determine installable from conflicting package path (%1%)", conflictError.conflictingPkgPath);
+                    return fallbackInstallable;
+                }
+                auto conflictingStorePath = *maybeConflictingStorePath;
+                auto conflictingManifestElement = find_if(manifest.elements.begin(), manifest.elements.end(), [&](const ProfileElement & currentProfileElement) {
+                    return currentProfileElement.storePaths.contains(conflictingStorePath);
+                });
+                if (conflictingManifestElement == manifest.elements.end()) {
+                    printError("warning: unable to find profile element for conflicting store path (%1%)", conflictingStorePath.to_string());
+                    return fallbackInstallable;
+                }
+                if (!conflictingManifestElement->source) {
+                    printError("warning: conflicting store path does not have a source (%1%)", conflictingStorePath.to_string());
+                    return fallbackInstallable;
+                }
+                return conflictingManifestElement->source->originalRef.to_string();
+            }();
+
+            throw Error(
+                "An existing package already provides the following file to the user's profile:\n"
+                "\n"
+                "  %1%\n"
+                "\n"
+                "This file conflicts with a file from the new package:\n"
```

Comment:
Here it's not 100% clear if "this" is referring to the one below or the one above, so I would reword as:

```suggestion
                "This is the conflicting file from the new package:\n"
```